lib/net/ldap/connection.rb

```diff
@@ -8,11 +8,11 @@ class Net::LDAP::Connection #:nodoc:
 
   def initialize(server)
     @instrumentation_service = server[:instrumentation_service]
-    server[:hosts] = [[server[:host], server[:port]]] if server[:hosts].nil?
 
     if server[:socket]
       prepare_socket(server)
     else
+      server[:hosts] = [[server[:host], server[:port]]] if server[:hosts].nil?
```

Comment:
Note: This works even when `server[:host]` and/or `server[:port]` are `nil`. It raises a `SocketError` and matches up with the old behavior.